core/src/reporter/state.ts

```diff
@@ -294,35 +316,31 @@ export class State {
         throw new OraklError(OraklErrorCode.WalletNotActive, msg)
       }
 
+      // Assumption: the only source of error can be json-rpc call
+      // Solution: keep polling/retrying until json-rpc becomes responsive
+      // If successful, the "break" statement will break the infinite loop
+      let retryCount = 0
       let remoteNonce: number
-      try {
-        if (this.delegatedFee) {
-          const caverWallet = wallet as CaverWallet
-          remoteNonce = Number(
-            await caverWallet.caver.rpc.klay.getTransactionCount(caverWallet.address)
-          )
-        } else {
-          remoteNonce = await (wallet as NonceManager).getTransactionCount()
+      while (true) {
+        try {
+          remoteNonce = await this.getRemoteNonce(wallet)
+          break
+        } catch (error) {
+          // Slack the error message every half an hour
```

Comment:
@Intizar-T Where does the `half an hour` comes from? Is it correct?

Reply:
It is better to remove such comments since they become outdated as soon as externally defined constant is updated.